codegen/s3.rs

```diff
@@ -1948,7 +1948,18 @@ struct KeyMarkerParser;
 impl KeyMarkerParser {
 	fn parse_xml<'a, T: Peek + Next>(tag_name: &str, stack: &mut T) -> Result<KeyMarker, XmlParseError> {
 		try!(start_element(tag_name, stack));
-		let obj = try!(characters(stack));
+		let mut obj = KeyMarker::default();
+
+		// if the contents are blank, we're done.
+		if peek_at_name(stack).unwrap() == "" {
```

Comment:
I think this needs to be refactored to look like the others (don't peek an stack.next() ).